### PR TITLE
Qaoa example

### PR DESCRIPTION
**Pull request by asciineuron**
# Added doc walking through Qaoa example using mps

### CODE REVIEW

**Comment by asciineuron** [comment]
Merged from main into this branch - should be good now. Only thing - the doctest fails because it is trying to run the qaoa notebook which takes longer than the max timeout period of 300 seconds (as expected). I am not sure if we could disable the timeout or how else to address this.

File: docs/qaoa_example.ipynb [hunk 1]
```diff
@@ -0,0 +1,465 @@
+{
```
Comment (rmlarose) [inline docs/qaoa_example.ipynb:1]: <p>Not sure why this output is here...</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/asciineuron/bgls/pull/21/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (rmlarose) [inline docs/qaoa_example.ipynb:1]: <p>Optional: Could mention same as the other MPS doc (and link to it)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/asciineuron/bgls/pull/21/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (rmlarose) [inline docs/qaoa_example.ipynb:1]: <p>Remove output</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/asciineuron/bgls/pull/21/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

**Comment by rmlarose** [comment]
Notebook looks great, thanks! Nice way to show off `run_sweep` functionality. Just a few minor comments to address.

It looks like the doctest on push finished in time and the doctest on pull_request timed out, so I'd guess it's right near the limit. I'd say it's fine to increase the execution timeout for now, it it's not too bad. Once we get tired of waiting there's better solutions we can do - substitute parameters when running on CI or, if possible, decrease runtime without sacrificing example quality.

**Pull request merged**